tests/loops/test_loops.py

```diff
@@ -956,8 +952,6 @@ def val_dataloader(self):
     # totals are increased by 1 (the failed batch which never completed)
     expected = state_dict.copy()
 
-    # TODO: `is_last_batch` is not correct on reload, the next line should not be necessary
```

Comment:
I remember being puzzled about why this did not work back when I wrote this.

Turns out the problem was that we were not restoring the `fetched` value on restart